src/Microsoft.AspNetCore.Http.Abstractions/Routing/IEndpointFeature.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Features
+{
+    /// <summary>
+    /// A feature interface for endpoint routing. Use <see cref="HttpContext.Features"/>
+    /// to access an instance associated with the current request.
+    /// </summary>
+    public interface IEndpointFeature
```

Comment:
This should be in the http.features assembly

Reply:
It depends on all the other types moved in this PR. If IEndpointFeature is in http.features then everything else is as well.

Yay/nay?

Reply:
Maybe, this is in 3.0 so I'm ok doing that and iterating. Session is similar.

Reply:
If Endpoint is in http.abstractions and not http.features then every future feature that uses it will need to be http.abstractions or higher (unless we type forward it again)